sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -236,6 +236,41 @@ public Object get(int ordinal) {
   }
 
   @Override
+  public Object get(int ordinal, DataType dataType) {
+    if (dataType instanceof NullType) {
```

Comment:
This reminds me: there's a potentially subtle pitfall with UnsafeRow if someone calls a primitive type-specific accessor without first checking the nullability: in that case, we'll currently return 0 instead of throwing an error.

I'm going to add some assertions to try to see if there are any places where we make this mistake.